threeML/catalogs/Fermi.py

```diff
@@ -1,58 +1,746 @@
 import numpy
+import re
 from VirtualObservatoryCatalog import VirtualObservatoryCatalog
 
 from astromodels import *
 from astromodels.utils.angular_distance import angular_distance
 
-from threeML.exceptions.custom_exceptions import custom_warnings
+from threeML.exceptions.custom_exceptions import custom_warnings, InvalidUTC
 from threeML.config.config import threeML_config
+from threeML.io.get_heasarc_table_as_pandas import get_heasarc_table_as_pandas
+
+import astropy.time as astro_time
+
+
+
+
+_trigger_name_match=re.compile("^(bn|grb?) ?(\d{9})$")
 
 class FermiGBMBurstCatalog(VirtualObservatoryCatalog):
+    def __init__(self, update=False):
+        """
+        The Fermi GBM GRB catalog. Search for GRBs  by trigger
+        number, location, spectral parameters, T90, and date range.
 
-    def __init__(self):
+        :param update: force update the XML VO table
+        """
 
         super(FermiGBMBurstCatalog, self).__init__('fermigbrst',
                                                    threeML_config['catalogs']['Fermi']['GBM burst catalog'],
-                                              'Fermi/GBM burst catalog')
+                                                   'Fermi/GBM burst catalog')
 
-    def apply_format(self, table):
+        self._gbm_detector_lookup = np.array(['n0', 'n1', 'n2', 'n3', 'n4', 'n5', 'n6',
+                                              'n7', 'n8', 'n9', 'na', 'nb', 'b0', 'b1'])
+
+        self._available_models = ('band', 'comp', 'plaw', 'sbpl')
+
+        self._vo_dataframe, self._all_table = get_heasarc_table_as_pandas('fermigbrst',
+                                                                          update=update,
+                                                                          cache_time_days=1.)
 
+
+
+    def apply_format(self, table):
         new_table = table['name',
                           'ra', 'dec',
                           'trigger_time',
-                          'Search_Offset']
+                          't90',
+
+        ]
 
         new_table['ra'].format = '5.3f'
         new_table['dec'].format = '5.3f'
 
-        return new_table.group_by('Search_Offset')
+        return new_table.group_by('trigger_time')
+
+    def search_trigger_name(self, *trigger_names):
```

Comment:
Ok, this has changed except for a few things. I kept a general separate query for source names because it saves a lot of typing. A also left a specific query in Swift to save typing. Other than that, pandas handles everything.